presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -173,6 +175,23 @@ public boolean canReplicate()
         PUSH_THROUGH_LOW_MEMORY_OPERATORS
     }
 
+    public enum PartitionMergingStrategy
+    {
+        LEGACY, // merge partition preference with parent but apply current partition preference
+        TOP_DOWN, // merge partition preference with parent and apply the merged partition preference
+        BOTTOM_UP; // don't merge partition preference and apply current partition preference only
+
+        public boolean mergePartitionPreferenceWithParent()
```

Comment:
nit: `isMergePartitionPreferenceWithParent`? Current name sounds like it is doing the "merge" step.